tests/conftest.py

```diff
@@ -524,3 +525,37 @@ def _maker(
         return cast(Path, config_file.relative_to(tmpdir_cwd))
 
     return _maker
+
+
+@pytest.fixture(scope="session")
+def setuptools_wheel_path(tmp_path_factory):
+    """
+    Ensure setuptools is downloaded and return the path to the download directory.
+    """
+    tmp_path = tmp_path_factory.mktemp("tmp_wheels")
+    subprocess.check_call(
+        [
+            sys.executable,
+            "-m",
+            "pip",
+            "download",
+            "--exists-action=i",
+            f"--dest={tmp_path.as_posix()}",
+            "setuptools",
+        ],
+        stdout=subprocess.DEVNULL,
+    )
+    return tmp_path
```

Comment:
Using a factory shouldn't be necessary.
```suggestion
def setuptools_wheel_path(tmp_path):
    """
    Ensure setuptools is downloaded and return the path to the download directory.
    """
    tmp_wheels_path = tmp_path / "tmp_wheels"
    tmp_wheels_path.mkdir()
    subprocess.check_call(
        [
            sys.executable,
            "-m",
            "pip",
            "download",
            "--exists-action=i",
            f"--dest={tmp_wheels_path.as_posix()}",
            "setuptools",
        ],
        stdout=subprocess.DEVNULL,
    )
    return tmp_wheels_path
```

Reply:
@shifqu oh, I didn't realized it's function-scoped. In that case, let's do two things:
1) Add a `session_tmp_path` fixture that is session-scoped, depends on the `tmp_path_factory` and behaves like `tmp_path` otherwise. And use that.
2) Add a code comment explaining the need.